# Week7 - Decision Tree Lab

* Train-test split
* Train a decison tree model
* Train a random forest model
* Evaluate the models
* Explain findings

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

df = pd.read_csv('https://raw.githubusercontent.com/msaricaumbc/DS_data/master/ds602/log_reg/employee-turnover-balanced.csv')
y = df['left_company']
X = df.iloc[:, 1:]

# Train-test split

In [6]:
from sklearn.model_selection import train_test_split
# split data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Train a decison tree model

In [35]:
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import accuracy_score, classification_report
# train decision tree classifier
dtc = DecisionTreeClassifier(random_state=42)
dtc.fit(X_train, y_train)
# make predictions on test set
dtc_pred = dt.predict(X_test)
# evaluate performance
print("Accuracy of decisionTreeClassifier:", accuracy_score(y_test, dtc_pred))
print(classification_report(y_test, dtc_pred))



Accuracy of decisionTreeClassifier: 0.835
              precision    recall  f1-score   support

          No       0.86      0.78      0.82        96
         Yes       0.81      0.88      0.85       104

    accuracy                           0.83       200
   macro avg       0.84      0.83      0.83       200
weighted avg       0.84      0.83      0.83       200



# Train a random forest model

In [37]:
from sklearn.ensemble import RandomForestClassifier
# train random forest classifier
rfc = RandomForestClassifier(n_estimators=100, random_state=42)
rfc.fit(X_train, y_train)

# make predictions on test set
y_pred = rfc.predict(X_test)

# evaluate performance
print("Accuracy of RandomForestClassifier:", accuracy_score(y_test, y_pred))
print(classification_report(y_test, y_pred))

Accuracy of RandomForestClassifier: 0.895
              precision    recall  f1-score   support

          No       0.88      0.91      0.89        96
         Yes       0.91      0.88      0.90       104

    accuracy                           0.90       200
   macro avg       0.89      0.90      0.89       200
weighted avg       0.90      0.90      0.90       200



# Evaluate the models

In [39]:
from sklearn.metrics import accuracy_score, classification_report

# evaluate decision tree model
y_pred_dt = dt.predict(X_test)
acc_dt = accuracy_score(y_test, y_pred_dt)
print("Decision Tree Accuracy:", acc_dt)
print(classification_report(y_test, y_pred_dt))

# evaluate random forest model
y_pred_rf = rfc.predict(X_test)
acc_rf = accuracy_score(y_test, y_pred_rf)
print("Random Forest Accuracy:", acc_rf)
print(classification_report(y_test, y_pred_rf))

Decision Tree Accuracy: 0.835
              precision    recall  f1-score   support

          No       0.86      0.78      0.82        96
         Yes       0.81      0.88      0.85       104

    accuracy                           0.83       200
   macro avg       0.84      0.83      0.83       200
weighted avg       0.84      0.83      0.83       200

Random Forest Accuracy: 0.895
              precision    recall  f1-score   support

          No       0.88      0.91      0.89        96
         Yes       0.91      0.88      0.90       104

    accuracy                           0.90       200
   macro avg       0.89      0.90      0.89       200
weighted avg       0.90      0.90      0.90       200



# Explain Findings


The accuracy of the decision tree model is 0.835, as can be seen from the outputs above, whereas the accuracy of the random forest model is 0.895. Hence, by comparing their accuracy scores, we may conclude that the Random Forest model is superior to the Decision Tree model.